### PR TITLE
test(hx-progress): unit test cases surf2017

### PR DESCRIPTION
**Pull request by lalithkarikelli**
## Description

Unit Test `<hx-progress>`

image

### What are the relevant story cards/tickets? Any additional PRs or other references?

Jira: SURF - 2017(Unit Test Cases)

## Before you request a review for this PR:

- [x] Did you add unit tests for any new code if needed?
- [x] Did you run the unit tests via `yarn test` to ensure they all pass?
- [x] Did you update the demo page and documentation if you changed/added functionality?
- [x] If needed, did you add or modify the demo test page to test the changed/added functionality?
- [x] Did you assign reviewers?
- [x] Have you linked to this PR from the Jira ticket(s)?

### CODE REVIEW

File: src/elements/hx-progress/index.spec.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { fixture, expect } from '@open-wc/testing';
```
Comment (100stacks) [inline src/elements/hx-progress/index.spec.js:1]: Please see PR #685 for updated testing criteria.

File: src/elements/hx-progress/index.spec.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+/**
+ * <hx-progress> component tests
+ *
+ * @type HXProgressElement
+ *
+ */
+describe('<hx-progress> tests', () => {
```
Comment (100stacks) [inline src/elements/hx-progress/index.spec.js:9]: Please add a screenshot of the test run in the `Description` block.

File: src/elements/hx-progress/index.spec.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    it(`rendered Light DOM should NOT equal simple template ${template}`, async () => {
+        const el = /** @type { HXProgressElement } */ await fixture(template);
+
+        expect(el).lightDom.to.not.equal(template);
+    });
+
+    it('should have a static Shadow DOM', async function () {
+        const el = /** @type { HXProgressElement } */ await fixture(template);
+        const sd = el.shadowRoot.innerHTML; // FIXME: temp fix issue submitted https://github.com/open-wc/open-wc/issues/1540
```
Comment (100stacks) [inline src/elements/hx-progress/index.spec.js:34]: This comment can be removed.

File: src/elements/hx-progress/index.spec.js (hunk 2 above)
Comment (lalithkarikelli) [reply src/elements/hx-progress/index.spec.js:9]: Changes have been done, please review.

File: src/elements/hx-progress/index.spec.js (hunk 3 above)
Comment (lalithkarikelli) [reply src/elements/hx-progress/index.spec.js:34]: Changes have been done, please review.

File: src/elements/hx-progress/index.spec.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        });
+
+        it(`the rendered light DOM should NOT equal simple template ${template}`, async () => {
+            const component = /** @type {HXProgressElement} */ await fixture(template);
+
+            expect(component).lightDom.to.not.equal(template);
+        });
+    });
+
+    describe('test $onConnect method', () => {
```
Comment (100stacks) [inline src/elements/hx-progress/index.spec.js:34]: @lalithkarikelli good work here.  👍 

Also, please add a test to check for default attribute `role`.  And another test to verify that the value of `role` is `progressbar`.  Please see example below.

File: src/elements/hx-progress/index.spec.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    });
+
+    describe('test <hx-progress> getter and setter methods', () => {
+        it('should be able to pause element', async () => {
+            const component = /** @type {HXProgressElement} */ await fixture(template);
+            component.paused = true;
+            const attr = component.hasAttribute('value');
+
+            expect(attr).to.be.true;
+        });
```
Comment (100stacks) [inline src/elements/hx-progress/index.spec.js:58]: Let's also add a test to adding and removing `paused`:

```js
        it('should be able to pause and unpause element', async () => {
            const component = /** @type {HXProgressElement} */ await fixture(template);
            component.paused = true;
            let attr = component.hasAttribute('paused');

            expect(attr).to.be.true;

            component.paused = false;
            attr = component.hasAttribute('paused');

            expect(attr).to.be.false;
        });
```

File: src/elements/hx-progress/index.spec.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        it(`the rendered light DOM should NOT equal simple template ${template}`, async () => {
+            const component = /** @type {HXProgressElement} */ await fixture(template);
+
+            expect(component).lightDom.to.not.equal(template);
+        });
+    });
+
+    describe('test $onConnect method', () => {
+        
+        it('should have aria-valuemin attribute', async () => {
```
Comment (100stacks) [inline src/elements/hx-progress/index.spec.js:36]: Let's also add a test to check the value of the `aria-valuemin` attribute, which should be  `0` using `component.getAttribute()`:

```js
        it('aria-valuemin should have a default value of 0', async () => {
            const defaultValue = 0;
            const component = /** @type {HXProgressElement} */ await fixture(template);
            const attr = component.hasAttribute('aria-valuemin');
            const value = component.getAttribute('aria-valuemin');
  
            expect(attr).to.be.true;
            expect(value).to.equal(defaultValue);
        });
```

File: src/elements/hx-progress/index.spec.js [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+            const component = /** @type {HXProgressElement} */ await fixture(template);
+            const attr = component.hasAttribute('aria-valuemin');
+
+            expect(attr).to.be.true;
+        });
+
+        it('should have aria-valuemax attribute', async () => {
```
Comment (100stacks) [inline src/elements/hx-progress/index.spec.js:43]: Let's also add a test to check the value of the `aria-valuemax` attribute, which should be `100`.  Please see example above.

File: src/elements/hx-progress/index.spec.js [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        it('should be able to pause element', async () => {
+            const component = /** @type {HXProgressElement} */ await fixture(template);
+            component.paused = true;
+            const attr = component.hasAttribute('value');
+
+            expect(attr).to.be.true;
+        });
+    });
+
+});
```
Comment (100stacks) [inline src/elements/hx-progress/index.spec.js:61]: Please add a blank line.

File: src/elements/hx-progress/index.spec.js (hunk 4 above)
Comment (100stacks) [inline src/elements/hx-progress/index.spec.js:34]: @lalithkarikelli good work here.  👍 

Also, please add a test to check for default attribute `role`.  And another test to verify that the value of `role` is `progressbar`.  Please see example below.

File: src/elements/hx-progress/index.spec.js (hunk 5 above)
Comment (lalithkarikelli) [reply src/elements/hx-progress/index.spec.js:94]: Changes have been done, please review.

File: src/elements/hx-progress/index.spec.js (hunk 6 above)
Comment (lalithkarikelli) [reply src/elements/hx-progress/index.spec.js:36]: Changes have been done, please review.

File: src/elements/hx-progress/index.spec.js (hunk 7 above)
Comment (lalithkarikelli) [reply src/elements/hx-progress/index.spec.js:53]: Changes have been done, please review.

File: src/elements/hx-progress/index.spec.js (hunk 8 above)
Comment (lalithkarikelli) [reply src/elements/hx-progress/index.spec.js]: Changes have been done, please review.

File: src/elements/hx-progress/index.spec.js (hunk 4 above)
Comment (lalithkarikelli) [reply src/elements/hx-progress/index.spec.js:34]: Changes have been done, please review.

File: src/elements/hx-progress/index.spec.js [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+        it('should have role attribute', async () => {
+            const component = /** @type {HXProgressElement} */ await fixture(template);
+            const attr = component.hasAttribute('role');
+
+            expect(attr).to.be.true;
+        });
+
+        it('should have role attribute set to progressbar', async () => {
+            const defaultValue = 'progressbar';
+            const component = /** @type {HXBusyElement} */ await fixture(template);
```
Comment (100stacks) [inline src/elements/hx-progress/index.spec.js:79]: Please change inline comment to:

```js
/** @type {HXProgressElement} */ 
```

File: src/elements/hx-progress/index.spec.js [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+            expect(attr).to.be.true;
+            expect(value).to.equal(String(attributeValue));
+        });
+
+    });
+
+    describe('test Shadow DOM', () => {
+        describe('verify render', () => {
+            it('should have a static Shadow DOM', async function () {
+                const component = /** @type { HXErrorElement } */ await fixture(template);
```
Comment (100stacks) [inline src/elements/hx-progress/index.spec.js:111]: Please change inline comment to:

```js
/** @type {HXProgressElement} */ 
```

File: src/elements/hx-progress/index.spec.js [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+                const shadow = component.shadowRoot.innerHTML;
+
+                expect(component).shadowDom.to.equal(shadow);
+            });
+
+            it('should render the Shadow Root mode open', async () => {
+                const component = /** @type { HXErrorElement } */ await fixture(template);
```
Comment (100stacks) [inline src/elements/hx-progress/index.spec.js:118]: Please change inline comment to:

```js
/** @type {HXProgressElement} */ 
```

File: src/elements/hx-progress/index.spec.js [hunk 12]
```diff
@@ -0,0 +1,139 @@
...
+
+                expect(mode).to.equal("open");
+            });
+
+        });
+
+        describe('verify Shadow DOM markup', () => {
+            it('markup should contain a #hxFill <span>', async () => {
+                const spanId = 'hxFill';
+                const component = /** @type { HXErrorElement } */ await fixture(template);
```
Comment (100stacks) [inline src/elements/hx-progress/index.spec.js:129]: Please change inline comment to:

```js
/** @type {HXProgressElement} */ 
```

File: src/elements/hx-progress/index.spec.js (hunk 9 above)
Comment (lalithkarikelli) [reply src/elements/hx-progress/index.spec.js]: Changes have been done, please review.

File: src/elements/hx-progress/index.spec.js (hunk 10 above)
Comment (lalithkarikelli) [reply src/elements/hx-progress/index.spec.js]: Changes have been done, please review.

File: src/elements/hx-progress/index.spec.js (hunk 11 above)
Comment (lalithkarikelli) [reply src/elements/hx-progress/index.spec.js]: Changes have been done, please review.

File: src/elements/hx-progress/index.spec.js (hunk 12 above)
Comment (lalithkarikelli) [reply src/elements/hx-progress/index.spec.js]: Changes have been done, please review.

**Pull request merged**